packages/jest-util/src/CustomConsole.js

```diff
@@ -41,11 +55,25 @@ export default class CustomConsole extends Console {
     super.log(message);
   }
 
+  _storeInBuffer(message: string, type: LogType) {
+    const callsite = getCallsite(3, this._getSourceMaps());
+    const origin = callsite.getFileName() + ':' + callsite.getLineNumber();
+
+    this._buffer.push({message, origin, type});
+    // we might want to keep this in the future for reporters (https://github.com/facebook/jest/issues/6441)
```

Comment:
I have this sorta working locally as well. The issue I ran into is that currently we implement `--verbose` and `--silent` by injecting the correct logger, which either logs eagerly (`verbose`), lazily (normal) or not at all (`silent`). If all loggers still put stuff into the buffer, we have to move that implementation to the reporters instead. Which makes sense, but is a larger change

Reply:
Not related anymore. I wanna overhaul reporters though (to support this, Ink, and more). Maybe for 25